underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystem.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.b2;
+
+import static com.backblaze.b2.client.contentSources.B2Headers.USER_AGENT;
+import static java.util.Objects.requireNonNull;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.PropertyKey;
+import alluxio.retry.RetryPolicy;
+import alluxio.underfs.ObjectUnderFileSystem;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.options.OpenOptions;
+import alluxio.util.UnderFileSystemUtils;
+import alluxio.util.io.PathUtils;
+
+import com.backblaze.b2.client.B2ListFilesIterable;
+import com.backblaze.b2.client.B2StorageClient;
+import com.backblaze.b2.client.B2StorageClientFactory;
+import com.backblaze.b2.client.exceptions.B2Exception;
+import com.backblaze.b2.client.structures.B2AccountAuthorization;
+import com.backblaze.b2.client.structures.B2Allowed;
+import com.backblaze.b2.client.structures.B2Bucket;
+import com.backblaze.b2.client.structures.B2FileVersion;
+import com.backblaze.b2.client.structures.B2ListFileNamesRequest;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ServiceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * B2 {@link UnderFileSystem} implementation based on the b2-sdk-core library.
+ */
+@ThreadSafe public class B2UnderFileSystem extends ObjectUnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(B2UnderFileSystem.class);
+
+  public static final String B2_SCHEME = "b2://";
+  private static final String FOLDER_SUFFIX = "/";
+  private static final String FOLDER_FILE_INDICATOR = ".bzEmpty";
+
+  private final B2StorageClient mB2StorageClient;
+  private final String mBucketName;
+  private final String mAccountId;
+  private final String mBucketId;
+  private final short mBucketMode;
+
+  /**
+   * Constructs a new instance of {@link B2UnderFileSystem}.
+   *
+   * @param alluxioURI the {@link AlluxioURI} for this UFS
+   * @param conf       the configuration for this UFS
+   * @return the created {@link B2UnderFileSystem} instance
+   * @throws ServiceException
```

Comment:
Most of the time, we don't need to javadoc the exceptions, especially since there isn't any important description for this exception. You can just remove this line. This may apply to other exception javadoc lines.

Reply:
Fixed